jaxrs/src/main/java/org/jboss/aerogear/unifiedpush/rest/registry/installations/InstallationRegistrationEndpoint.java

```diff
@@ -217,17 +222,28 @@ public Response registerInstallation(
     @Produces(MediaType.APPLICATION_JSON)
     @ReturnType("org.jboss.aerogear.unifiedpush.rest.EmptyJSON")
     public Response increasePushMessageReadCounter(@PathParam("id") String pushMessageId,
-                           @Context HttpServletRequest request) {
+                           @Context HttpServletRequest request) throws IOException {
 
         // find the matching variation:
         final Variant variant = loadVariantWhenAuthorized(request);
         if (variant == null) {
             return create401Response(request);
         }
 
-        //let's do update the analytics
         if (pushMessageId != null) {
+            try (InputStream props = this.getClass().getResourceAsStream("/kafka/producer.props")) {
+                Properties properties = new Properties();
```

Comment:
I think we should use the `final` keyword here, to indicate that the `Properties` and `Producer` are immutable